src/DoctrineObject.php

```diff
@@ -351,6 +350,7 @@ protected function hydrateByValue(array $data, $object)
 
             if ($metadata->hasAssociation($field)) {
                 $target = $metadata->getAssociationTargetClass($field);
+                assert($target !== null);
```

Comment:
This is a sign that we should have a method that throws instead of returning `null`

Reply:
Indeed, yes :-)

Reply:
Above assert code is failing for mongoDB, when we are using EmbedMany argument without target document locally.

According to documentation, EmbedMany without target document is allowed to embed different documents in same field.(https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/2.7/reference/embedded-mapping.html#mixing-document-types)

Does this assert  check is really needed as null value is also accepted for mongodb?